x-pack/plugins/monitoring/server/config.ts

```diff
@@ -227,4 +73,23 @@ export const configSchema = schema.object({
   }),
 });
 
-export type MonitoringConfig = TypeOf<typeof configSchema>;
+class MonitoringElasticsearchConfig extends ElasticsearchConfig {
+  public readonly logFetchCount?: number;
+
+  constructor(rawConfig: TypeOf<typeof monitoringElasticsearchConfigSchema>) {
+    super(rawConfig as ElasticsearchConfigType);
+    this.logFetchCount = rawConfig.logFetchCount;
+  }
+}
+
+export type MonitoringConfig = ReturnType<typeof createConfig>;
+export function createConfig(config: TypeOf<typeof configSchema>) {
+  return {
+    ...config,
+    elasticsearch: new MonitoringElasticsearchConfig(config.elasticsearch),
```

Comment:
I think this could just be:
```suggestion
    elasticsearch: new ElasticsearchConfig(config.elasticsearch),
```

Since there is no `monitoring.elasticsearch.logFetchCount`, right?

Reply:
Done